cc.cmake assumes GCC_VERSION, causing build failure with NVHPC

Hello, 

I am trying to build OpenBLAS with NVHPC on anIntel Sapphire Rapids system, and I am running into issues in `cc.cmake`.

With this setup, the following section is triggered: 
https://github.com/OpenMathLib/OpenBLAS/blob/02dc625d42ad3ccb61d2e47c073741614c01dea6/cmake/cc.cmake#L188-L198

However, when building with NVHPC, $GCC_VERSION is not defined. As a result CMake fails with an error like this : 
```
CMake Error at cmake/cc.cmake:191 (if):
  if given arguments:

    "VERSION_GREATER" "11.0" "OR" "VERSION_EQUAL" "11.0"

  Unknown arguments specified
```
It seems that this condition assumes a GNU compiler, and should probably only be evaluated when that is the case.

I believe similar issues may occur for other microarchitectures, such as:
* cooperlake : https://github.com/OpenMathLib/OpenBLAS/blob/02dc625d42ad3ccb61d2e47c073741614c01dea6/cmake/cc.cmake#L179
* zen : https://github.com/OpenMathLib/OpenBLAS/blob/02dc625d42ad3ccb61d2e47c073741614c01dea6/cmake/cc.cmake#L204
* a64fx when disabling SVE : https://github.com/OpenMathLib/OpenBLAS/blob/02dc625d42ad3ccb61d2e47c073741614c01dea6/cmake/cc.cmake#L218
etc.

Whike looking further into the file, I also noticed that `$GCC_VERSION` is set even when the compiler is Intel LLVM :
https://github.com/OpenMathLib/OpenBLAS/blob/02dc625d42ad3ccb61d2e47c073741614c01dea6/cmake/cc.cmake#L36

This might contribute to inconsistent behavior accross compiler, and most likely implies missuse of this variable (testing against this value) somewhere else in the CMake files.

Best.

--- a/cmake/cc.cmake
+++ b/cmake/cc.cmake
@@ -113,6 +113,7 @@ if (${CMAKE_C_COMPILER_ID} STREQUAL "PGI")
 endif ()
 
 if (${CMAKE_C_COMPILER_ID} STREQUAL "NVHPC")
+  set (GCC_VERSION 100)
   if (POWER)
     set(CCOMMON_OPT "${CCOMMON_OPT} -tp pwr8")
   elseif (X86_64)

